src/fosslight_util/download.py

```
@@ -880,6 +892,82 @@ def download_file(url, target_dir):
+
+    Tries ``rpm2cpio | cpio -idmv`` (common on Linux), then ``bsdtar``/``tar`` if
+    libarchive-backed tar can read the RPM.
+    """
+    abs_src = os.path.abspath(source_file)
+    dest_path = os.path.abspath(dest_path)
+    Path(dest_path).mkdir(parents=True, exist_ok=True)
+
+    rpm2cpio = shutil.which("rpm2cpio")
+    cpio = shutil.which("cpio")
+    if rpm2cpio and cpio:
+        p1 = subprocess.Popen(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)